cpp/ql/src/Security/CWE/CWE-611/XXE.ql

```diff
@@ -219,6 +255,23 @@ class XXEConfiguration extends DataFlow::Configuration {
         call.getThisArgument() and
       encodeXercesFlowState(flowstate, 0, 1) // default configuration
     )
+    or
+    // source is an `options` argument on a `libxml2` parse call that specifies
```

Comment:
I feel like every new XML library adds a source/sink in the dataflow configuration. It might be a better design to have an abstract class ("XmlLibrary" or whatever) in the query that can provide:
- The transformer
- The source
- The sink
We don't need to do this in this PR, but it might be a good refactoring to do at some point.

Reply:
I agree.  I've had similar thoughts, and I also want to move the library-specific code out into one or more `.qll` library files at some point to make the query more readable - a change like this will enable that.

Added to the TODO list on the issue.